### PR TITLE
NT-976, NT-980, NT-978, NT-979 - Approval for mobile

### PR DESCRIPTION
**Pull request by eiriko2303**

### CODE REVIEW

File: src/pages/approval/approval-pm/approval-pm.html [hunk 1]
```diff
@@ -34,18 +34,39 @@ <h5 class="username">{{user.name.split(',').reverse().join(' ')}}</h5>
...
                 [withActionButtons]="true"
                 [isPM]="true"
                 (onApproveOneClick)="approveLine($event)"
                 (onCloseClick)="setOpen(project.no, user.no, period.startDate)"
                 (onMoreOptionsClick)="showMoreOptions($event.entry, $event.clickEvent)"
-              ></approval-entries-grid>
+                ></approval-entries-grid>
+              </div>
+              <div *ngIf="isPeriodOpen(project.no, user.no, period.startDate)" class="mobile-only">
+                  <approval-entries-mobile [entries]="getPeriodEntriesForUserInProject(project.no, user.no, period.startDate)" [isPM]="true"></approval-entries-mobile>
```
[the remark this answers is not in the document]
Comment (eiriko2303) [reply src/pages/approval/approval-pm/approval-pm.html]: Yep that is true :)  ok i will rename it in the branch that you approved earlier which is with the introduction of approval-entries-mobile and then merge it again into this branch ;)

[the remark this answers is not in the document]
Comment (eiriko2303) [reply src/pages/approval/approval-pm/approval-pm.html]: Have changed it now in both branches (NT-978 and NT-979) 😄

File: src/app/app.component.ts [hunk 2]
```diff
@@ -124,6 +130,21 @@ export class MyApp {
     this.keyboard$.subscribe(show => (this.showingKeyboard = show));
   }
 
+  setApprovalEntries() {
```
[the remark this answers is not in the document]
Comment (eiriko2303) [reply src/app/app.component.ts]: Yep I agree, gonna move them ;)

File: src/components/approval-entries-grid/approval-entries-grid.component.ts [hunk 3]
```diff
@@ -1,33 +1,33 @@
...
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+
+import { EntryForApprovalByPM } from '../../models';
+import { BaseEntryForApproval } from '../../models/baseEntryForApproval';
+
+@Component({
+    selector: 'approval-entries-grid',
+    templateUrl: 'approval-entries-grid.component.html',
+    animations: [
+      trigger(
```
[the remark this answers is not in the document]
Comment (eiriko2303) [reply src/components/approval-entries-grid/approval-entries-grid.component.ts]: Fixed :)

File: src/pages/approval/approval-lm/approval-lm.html [hunk 4]
```diff
@@ -35,24 +35,54 @@ <h5 class="username">{{user.name.split(',').reverse().join(' ')}}</h5>
                 </ion-col>
                 <ion-col col-4><div class="open-arrow"></div></ion-col>
               </ion-row>
+              <div class="period-header-summary-mobile period-header-summary-mobile-LM mobile-only">
+                <div class="user-container user-container-mobile-LM">
+                  <img src="assets/images/User.svg" class="user-picture" />
+                  <h5 class="username">{{user.name.split(',').reverse().join(' ')}}</h5>
```
[the remark this answers is not in the document]
Comment (eiriko2303) [reply src/pages/approval/approval-lm/approval-lm.html]: Agree, gonna move it to the formattingservice :)

File: src/pages/approval/approval.html [hunk 5]
```diff
@@ -35,12 +37,13 @@
...
-    [periodHoursSummary]="periodHoursSummary"></approval-lm>
+    [periodHoursSummary]="periodHoursSummary"
+    [isWeb]="isWeb"></approval-lm>
   </ng-container>
 </ion-content>
 
-<ion-footer id="footer-scroll" class="footer-web">
-  <ion-grid class="footer-web-toolbar-grid">
+<ion-footer id="footer-scroll" class="footer-web" *ngIf="isWeb">
+  <ion-grid class="footer-web-toolbar-grid mobile-only">
```
[the remark this answers is not in the document]
Comment (eiriko2303) [reply src/pages/approval/approval.html]: Yep that was wierd, gonna remove mobile-only as this footer is just on web.

File: src/pages/reject-hours/reject-hours.html [hunk 6]
```diff
@@ -1,38 +1,39 @@
...
       </ion-title>
     </ion-toolbar>
+    <top-toolbar *ngIf="!isWeb" [titleText]="'Reject hours'"></top-toolbar>
     <save-animation></save-animation>
   </ion-header>
-  
-  <ion-content>
-    <div class="user-container">
+
+  <ion-content [ngClass]="{'mobile' : !isWeb}">
```
[the remark this answers is not in the document]
Comment (eiriko2303) [reply src/pages/reject-hours/reject-hours.html]: Yep, fixed it now, it was just the header that needed to stay like that :)

File: src/pages/approval/approval-pm/approval-pm.html [hunk 7]
```diff
@@ -34,18 +34,39 @@ <h5 class="username">{{user.name.split(',').reverse().join(' ')}}</h5>
...
+                      [customClass]="'approve-all-btn'"
+                      (actionClick)="approveAllEntriesInPeriod(project.no, user.no, period.startDate)"></action-button>
+                    </div>
+                    <div class="open-arrow"></div>
+                  </div>
+                </div>
               </div>
-              <approval-entries-grid *ngIf="isPeriodOpen(project.no, user.no, period.startDate)"
-                [entries]="getPeriodEntriesForUserInProject(project.no, user.no, period.startDate)"
+              <div *ngIf="isPeriodOpen(project.no, user.no, period.startDate)" class="web-only">
```
[the remark this answers is not in the document]
Comment (eiriko2303) [reply src/pages/approval/approval-pm/approval-pm.html]: Ok, will fix that :)

File: src/components/change-manager-popover/change-manager-popover.component.ts [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+})
+export class ChangeManagerPopoverComponent {
+    approveAsPM: boolean;
+    constructor(public navParams: NavParams,
+                public viewCtrl: ViewController) {
+        this.approveAsPM = navParams.data.approveAsPM;
+    }
+
+    ionViewWillEnter() {
+        const popoverContent = document.getElementsByClassName('popover-content')[0] as HTMLDivElement;
```
[the remark this answers is not in the document]
Comment (eiriko2303) [reply src/components/change-manager-popover/change-manager-popover.component.ts]: What do you think about this @noro5 ? :)

[the remark this answers is not in the document]
Comment (reg0) [reply src/components/change-manager-popover/change-manager-popover.component.ts]: in host attribute you can add some class on top level element, so then you can add css specific to that popover without affecting others, I did such thinkg with ```worktime-standard-page``` class in settings pages (my-devices.ts for example)

[the remark this answers is not in the document]
Comment (eiriko2303) [reply src/components/change-manager-popover/change-manager-popover.component.ts]: Acutally i did tried what you proposed, but the problem there was that this host attribute is adding classes to the component itself, but the problem for me was that i needed to add that to the parent of the component. But i did find a solution that Ionic had for this: 

let popover = this.popoverCtrl.create(ChangeManagerPopoverComponent, data, {cssClass: 'change-manager-popover-style'});

So with cssClass parameter you can add styling to the specific popover :)

**Pull request merged**